### PR TITLE
Many packages from SciPy sprints important to numpy, scipy, numba, matplotlib

### PR DESCRIPTION
**Pull request by costrouc**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by costrouc** [comment]
Needed for python downstream testing. Motivation in directory https://github.com/costrouc/python-downstream-testing

**Comment by FRidh** [comment]
I like what you've done in your repo. It's been on my mind to set up a Hydra instance to do the same things for numpy, scipy and pandas, but also core packages such as pip, setuptools and wheel.

**Comment by costrouc** [comment]
This is a great opportunity for us to get more attention to nix. Currently a lot of work is being out into doing this work with conda (however due to conda structure they are not able to test all downstream packages -- it's a manual process for them). Getting nix to build current master  and downstream packages of many repositories is really not that difficult. The only issue I've run into is that some nix derivations were configured to build the pypi tarbal not GitHub. Do you have any advice? What would the resources required be? Ive considered using Hercules CI and hydra. I see the build infrastructure being the harder part. I'm determined to get the POC working and know how to get this noticed once it is working.

**Comment by FRidh** [comment]
Initially, I think I would go for a Hydra instance, with a job per project. Each job could evaluate the same expressions, however, in each job one of the expressions would be overridden to point to master. We could eventually also look into combinations (numpy@master and scipy@master). Hydra could monitor the repo's, fetching e.g. every 5 minutes. I think a single quadcore machine would be more than sufficient.

If projects want to use it and make it part of their CI they could use Travis to perform a Nix build. The project would then need a simple Nix expression that injects the package in the package set and builds a list of packages.

**Comment by costrouc** [comment]
Thanks for the advice. I have a xeon quad core at home that's not doing much at the moment (and running nixos) so I'll setup hydra. I'm not too familiar with hydra (tired to set it up once). I think that will be the bigger learning curve for me.

File: pkgs/development/python-modules/drms/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    pandas
+    six
+  ];
+
+  checkInputs = [
+    pytest
+  ];
+
+  checkPhase = ''
+    python -m drms.tests
```
Comment (FRidh) [inline pkgs/development/python-modules/drms/default.nix]: ${python.interpreter}

**Review by FRidh — COMMENTED** [review]
Please don't make the PR *too* big.

File: pkgs/development/python-modules/drms/default.nix (hunk 1 above)
Comment (costrouc) [reply pkgs/development/python-modules/drms/default.nix]: Fixed. and stoping to add package haha. It's too easy with python-package-init

**Comment by costrouc** [comment]
@FRidh I am stopping to add packages. Right now working on getting nix-review to pass

**Comment by costrouc** [comment]
@FRidh this package is ready for review and merge. I know that it causes a significant number of rebuilds. Happy to rebase to staging if needed. I have testing that all packages build.

**Comment by FRidh** [comment]
The decorator bump may cause trouble. Well, let's hope it's OK.

**Pull request merged**